spec/models/user_spec.rb

```diff
@@ -2,7 +2,139 @@
 
 RSpec.describe User, type: :model do
   context 'basic functionality' do
-    it 'has a short_name' do
+    let(:user) { FactoryBot.build(:user)}
+    it 'has a valid factory' do
+      user.valid?
+      expect(user).to be_valid
     end
+    it 'is not valid without an id' do
+      user.id = nil
+      user.valid?
+      expect(user.errors.details).to include(:id => [{:error => :blank}])
+    end
+
+    it 'is valid with a name' do
+      ['', 'uno', 'dos tres'].each do |x|
+        user.name = x
+        user.valid?
+        expect(user).to be_valid
+      end
+    end
+
+    it 'is valid with a null name' do
+      user.name = nil
+      user.valid?
+      expect(user).to be_valid
+    end
+
+    it 'admin is true or false' do
+      [true, false].each do |x|
+        user.admin = x
+        user.valid?
+        expect(user).to be_valid
+      end
+
+      user.admin = nil
+      user.valid?
+      expect(user.errors.details).to include(admin: [{error: :inclusion, value: nil}])
+    end
+
+    it 'staff is true or false' do
+      [true, false].each do |x|
+        user.staff = x
+        user.valid?
+        expect(user).to be_valid
+      end
+
```

Comment:
Spec about if is admin is a staff so admin:true staff:true is ok and admin:true staff:false too